File: ai/domain/news/model/news_article.py
from pydantic import BaseModel
from typing import Optional


class NewsArticle(BaseModel):
    """뉴스 기사 모델"""
    type: str
    title: str
    date: str
    image: str
    link: str
    description: str
    
    class Config:
        json_schema_extra = {
            "example": {
                "type": "경제",
                "title": "뉴스 제목",
                "date": "2025.01.30",
                "image": "https://example.com/image.jpg",
                "link": "https://example.com/news/123",
                "description": "뉴스 설명"
            }
        }

File: ai/spring/news/src/main/java/kr/yeotaeho/api/dto/NewsArticle.java
package kr.yeotaeho.api.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class NewsArticle {
    private String type;
    private String title;
    private String date;
    private String image;
    private String link;
    private String description;
}

